addons/crm_partner_assign/wizard/crm_forward_to_partner.py

```diff
@@ -112,73 +94,56 @@ def action_forward(self, cr, uid, ids, context=None):
 
-    _columns = {
-        'forward_type': fields.selection([('single', 'a single partner: manual selection of partner'), ('assigned', "several partners: automatic assignation, using GPS coordinates and partner's grades"), ], 'Forward selected leads to'),
-        'partner_id': fields.many2one('res.partner', 'Forward Leads To'),
-        'assignation_lines': fields.one2many('crm.lead.assignation', 'forward_id', 'Partner Assignation'),
-        'body': fields.html('Contents', help='Automatically sanitized HTML contents'),
-    }
 
-    _defaults = {
-        'forward_type': lambda self, cr, uid, c: c.get('forward_type') or 'single',
-    }
-
-
-class crm_lead_assignation (osv.TransientModel):
+class crm_lead_assignation (models.TransientModel):
     _name = 'crm.lead.assignation'
-    _columns = {
-        'forward_id': fields.many2one('crm.lead.forward.to.partner', 'Partner Assignation'),
-        'lead_id': fields.many2one('crm.lead', 'Lead'),
-        'lead_location': fields.char('Lead Location', size=128),
-        'partner_assigned_id': fields.many2one('res.partner', 'Assigned Partner'),
-        'partner_location': fields.char('Partner Location', size=128),
-        'lead_link': fields.char('Lead  Single Links', size=128),
-    }
-
-    def on_change_lead_id(self, cr, uid, ids, lead_id, context=None):
-        if not context:
-            context = {}
-        if not lead_id:
-            return {'value': {'lead_location': False}}
-        lead = self.pool.get('crm.lead').browse(cr, uid, lead_id, context=context)
+    forward_id = fields.Many2one('crm.lead.forward.to.partner', string='Partner Assignation')
+    lead_id = fields.Many2one('crm.lead', string='Lead')
+    lead_location = fields.Char(string='Lead Location', size=128)
+    partner_assigned_id = fields.Many2one('res.partner', string='Assigned Partner')
+    partner_location = fields.Char(string='Partner Location', size=128)
+    lead_link = fields.Char(string='Lead  Single Links', size=128)
+    
+    @api.onchange('lead_id')
+    def on_change_lead_id(self):
         lead_location = []
-        if lead.country_id:
-            lead_location.append(lead.country_id.name)
-        if lead.city:
-            lead_location.append(lead.city)
-        return {'value': {'lead_location': ", ".join(lead_location)}}
-
-    def on_change_partner_assigned_id(self, cr, uid, ids, partner_assigned_id, context=None):
-        if not context:
-            context = {}
-        if not partner_assigned_id:
-            return {'value': {'lead_location': False}}
-        partner = self.pool.get('res.partner').browse(cr, uid, partner_assigned_id, context=context)
-        partner_location = []
-        if partner.country_id:
-            partner_location.append(partner.country_id.name)
-        if partner.city:
-            partner_location.append(partner.city)
-        return {'value': {'partner_location': ", ".join(partner_location)}}
+        if self.country_id:
```

Comment:
I am crm.lead.assignation and I don't have country_id field (and yes it applies to whole file)